### PR TITLE
CrossGen comparison - compare debugging files

### PR DESCRIPTION
**Pull request by echesakovMSFT**
This PR adds one more checking in crossgen_comparison scenario for
1. Output of "crossgen /CreatePerfMap" on Linux
2. Output of "crossgen /CreatePdb" on Windows

This will ensure that crossgen output is identical between different cross-architectures and between a cross-architecture and the native architecture for given set of input assemblies

### CODE REVIEW

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Ubuntu arm Cross Checked crossgen_comparison Build and Test

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Ubuntu arm Cross Checked crossgen_comparison Build and Test

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Ubuntu arm Cross Checked crossgen_comparison Build and Test

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Ubuntu arm Cross Checked crossgen_comparison Build and Test

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Ubuntu arm Cross Checked crossgen_comparison Build and Test

File: tests/scripts/crossgen_comparison.py [hunk 1]
```diff
@@ -409,35 +455,77 @@ def _decode_object(self, dict):
...
     with open(json_filename, 'wt') as json_file:
         json.dump(crossgen_result, json_file, cls=CrossGenResultEncoder, indent=2)
 
+def save_crossgen_results_to_json_files(crossgen_results, result_dirname):
+    for result in crossgen_results:
+        json_filename = os.path.join(result_dirname, "{0}-{1}.json".format(result.assembly_name, result.output_file_type))
+        save_crossgen_result_to_json_file(result, json_filename)
+
+def create_output_folders():
+    ni_files_dirname = tempfile.mkdtemp()
```
Comment (jashook) [inline tests/scripts/crossgen_comparison.py]: When is this going to be cleaned?

Comment (echesakovMSFT) [reply tests/scripts/crossgen_comparison.py]: Addressed

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Ubuntu arm Cross Checked crossgen_comparison Build and Test

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Ubuntu arm Cross Checked crossgen_comparison Build and Test

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Ubuntu arm Cross Checked crossgen_comparison Build and Test

**Comment by echesakovMSFT** [comment]
"crossgen_comparison" failed due to a regression introduced somewhere making results of Hostx64/arm crossgen for SPC different from "reference" Hostarm/arm crossgen. The test scenario per se works as expected - I am going to merge this and work on finding the regression commit.

/cc @RussKeldorph

**Pull request merged**